openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -323,20 +326,70 @@ def test_non_live_course(self):
         )
         self.assertRedirects(response, expected_url)
 
+    @override_waffle_flag(CONTENT_TYPE_GATING_FLAG, True)
+    @mock.patch.dict(settings.FEATURES, {'DISABLE_START_DATES': False})
+    def test_course_does_not_expire_for_different_roles(self):
+        """
+        There are a number of different roles/users that should not lose access after the expiration date.
+        Ensure that users who should not lose access get a 200 (ok) response
+        when attempting to visit the course after their would be expiration date.
+        """
+        course = CourseFactory.create(start=THREE_YEARS_AGO)
+        url = course_home_url(course)
+
+        # create a list of those users who should not lose their access,
+        # then assert that their access persists past the 'expiration date'
+        users_no_expired_access = []
+
+        verified_user = UserFactory(password=self.TEST_PASSWORD)
+        verified_enrollment = CourseEnrollment.enroll(verified_user, course.id, mode=CourseMode.VERIFIED)
+        ScheduleFactory(start=THREE_YEARS_AGO, enrollment=verified_enrollment)
+        users_no_expired_access.append((verified_user, 'Verified Learner'))
+
+        # There are two types of course team members: instructor and staff
+        # they have different privileges, but for the purpose of this test the important thing is that they should
+        # retain their access to the course after the access would expire for a normal audit learner
+        instructor = UserFactory.create(password=self.TEST_PASSWORD)
+        enrollment = CourseEnrollment.enroll(instructor, course.id, mode=CourseMode.AUDIT)
+        CourseInstructorRole(course.id).add_users(instructor)
+        ScheduleFactory(start=THREE_YEARS_AGO, enrollment=enrollment)
+        users_no_expired_access.append((instructor, 'Course Instructor'))
+
+        staff = UserFactory.create(password=self.TEST_PASSWORD)
```

Comment:
if we end up adding the other roles to this function I would probably suggest a loop, but this is fine for now